components/login/login_controller.jsx

```diff
@@ -75,6 +75,12 @@ export default class LoginController extends React.Component {
     preSubmit(e) {
         e.preventDefault();
 
+        const {location} = this.props;
+        const newQuery = location.search.replace(/(extra=password_change)&?/i, '');
+        if (newQuery !== location.search) {
+            browserHistory.replace(`${location.pathname}${newQuery}${location.hash}`);
```

Comment:
This doesn't cause any problems if removing the extra parameter would cause the link to end in `?` or `?something=else&` would it?

Reply:
@hmhealey Good eye. I didn't bother removing the trailing `?` or `&` when I was writing it because I'm not aware of it those characters trailing a path or query (respectively) causing any issues. If you're aware of any reason we should be concerned let me know and I can make a change.

Reply:
There's nothing I'm aware of, so this should be fine then